### PR TITLE
Upgrade Instructions.

### PR DESCRIPTION
**Pull request by mohitsoni**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -324,6 +326,23 @@ curl -v -H "Authorization: token=$(dcos config show core.dcos_acs_token)" -X POS
...
+
+1. Perform [Backup Operation](https://github.com/mesosphere/dcos-cassandra-service#backup) on your currently running Cassandra Service. Please make a note of the backup name and backup location.
+2. [Install a new Cassandra Service](https://github.com/mesosphere/dcos-cassandra-service#multiple-cassandra-cluster-installation) instance.
+3. Perform [Restore operation](https://github.com/mesosphere/dcos-cassandra-service#restore) on the new cluster created in Step #2
+4. Once the restore operation is finished, please check if the data is restored correctly.
+5. Optionally, [uninstall](https://github.com/mesosphere/dcos-cassandra-service#uninstall) old cluster.
+
+## 2. Multi-DC Replication assisted
+
+1. [Install a new Cassandra Service in Multi-DataCenter configuration](https://github.com/mesosphere/dcos-cassandra-service#multi-datacenter-deployments), so that if becomes a peer DataCenter of your original Cassandra Service
```
Comment (triclambert) [inline README.md]: s/if/it/

File: README.md [hunk 2]
```diff
@@ -324,6 +326,23 @@ curl -v -H "Authorization: token=$(dcos config show core.dcos_acs_token)" -X POS
...
+2. Cassandra Service created is step #1 will start replicating data from your original cluster, once it's online.
```
Comment (triclambert) [inline README.md]: s/is/in/

File: README.md [hunk 3]
```diff
@@ -324,6 +326,23 @@ curl -v -H "Authorization: token=$(dcos config show core.dcos_acs_token)" -X POS
...
+2. [Install a new Cassandra Service](https://github.com/mesosphere/dcos-cassandra-service#multiple-cassandra-cluster-installation) instance.
+3. Perform [Restore operation](https://github.com/mesosphere/dcos-cassandra-service#restore) on the new cluster created in Step #2
+4. Once the restore operation is finished, please check if the data is restored correctly.
+5. Optionally, [uninstall](https://github.com/mesosphere/dcos-cassandra-service#uninstall) old cluster.
+
+## 2. Multi-DC Replication assisted
+
+1. [Install a new Cassandra Service in Multi-DataCenter configuration](https://github.com/mesosphere/dcos-cassandra-service#multi-datacenter-deployments), so that if becomes a peer DataCenter of your original Cassandra Service
+2. Cassandra Service created is step #1 will start replicating data from your original cluster, once it's online. 
+3. Once data replication finishes, perform data validation.
```
Comment (triclambert) [inline README.md]: Didn't you say there's no way to tell when replication finishes?

File: README.md [hunk 4]
```diff
@@ -324,6 +326,23 @@ curl -v -H "Authorization: token=$(dcos config show core.dcos_acs_token)" -X POS
...
+4. Optionally, [uninstall](https://github.com/mesosphere/dcos-cassandra-service#uninstall) old cluster.
```
Comment (triclambert) [inline README.md]: Is this really optional?  Seems recommended; recommend dropping the optional word.

File: README.md (hunk 3 above)
Comment (mohitsoni) [inline README.md]: Good point. Let's make it more explicit. 

Basically, it's upto the operator to determine if the replication has finished or not.

**Comment by mohitsoni** [comment]
@mesosphere-ci please retest this.

**Comment by bernadinm** [comment]
Hi @mohitsoni,

We'll also need to inform the user that all the keyspaces and tables in Cassandra must need to exist prior to a new restore.    With the current backup and restore documentation, if they go though the restore, they'll notice it wont work for restores across different Cassandra clusters. The logs will say `Skipping file la-1-big-Data.db: table <INSERT_KEYSPACE_HERE>.<INSERT_TABLE_HERE> doesn't exist`. They would assume this would be created for them during restore which is not the case.

Another thing to note is that I couldn't get Cassandra [1.0.5-2.2.5] and [1.0.13-2.2.5] to run simultaneously. I had to uninstall [1.0.5-2.2.5] completely, create the keyspace, table and then finally perform the restore.

I have a questions from these findings:

1. Can we get the backup and restore to automatically create the keyspace and table? For example
`CREATE KEYSPACE IF NOT EXISTS` and `CREATE TABLE IF NOT EXISTS` should work but I am not sure how the backup and restore was implemented in this case. If the restore can create the keyspace and tables then we wont need to add it in the documentation.

2. If indeed that a user can't run both Cassandra [1.0.5-2.2.5] and [1.0.13-2.2.5] then we'll need to explicitly mention that the proper upgrade would be to remove the old C* cluster entirely and perform the proper restore steps on the new cluster. If you have some cycles, you can test this as well but I couldn't get both of them to work on my end even when cleaning out ZK and running janitor.py.

**Comment by bernadinm** [comment]
I got in touch with @gabrielhartmann and he mentioned that a different method needs to be performed for my question 1. At this point question 2 remains.

**Comment by mohitsoni** [comment]
@bernadinm Just curious to understand why you were not able to run both Cassandra [1.0.5-2.2.5] and [1.0.13-2.2.5] on one DC/OS cluster. What was the size of your DC/OS cluster (private nodes) ?

And, what errors did you see in logs?

**Comment by bernadinm** [comment]
@mohitsoni, I wish I got the logs from my previous test. However I did test again just now with 6 private agents and I see it did work this time. The last time I had 5. Do we need a minimum of 3 per C* instance? If so, maybe I can add this requirements to the "Pre-Install Notes" of the Cassandra package.json in the universe.

**Comment by mohitsoni** [comment]
@bernadinm Currently, we launch 1 Cassandra node per host/agent. The default cluster size of C* instance as configured in Universe is 3 instances, that's why you need 3 agents to run the default configuration.

**Pull request merged**